### ISSUE TITLE
Cannot debug on an iPad "Failed to start remote service "com.apple.debugserver" on device."

### ISSUE THREAD

**Issue by sowens-csd** [flutter/flutter#71044]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Try to run in debug mode on a device running iOS 14.1

**Expected results:** 
The app running on the device

**Actual results:** 
An error message
Failed to start remote service "com.apple.debugserver" on device.

<details>
  <summary>
The app didn't launch with Xcode 12.2 and iOS 14.1 on an iPad 6th gen. I was able to get it to work by upgrading to iOS 14.2 on the device. I'm submitting this report in case it helps someone else with a similar problem and because I've seen that the Flutter team has had similar issues resolved fairly recently. You can close this issue immediately if you'd like. I'm not waiting on it, only leave it open if you find it helpful. 

Here's the log output from Flutter:
```
Error launching app. Try launching from within Xcode via:
    open ios/Runner.xcworkspace

Your Xcode version may be too old for your iOS version.
═══════════════════════════════════════════════════════════════════════════════════
2020-11-22 12:08:46.683 ios-deploy[75249:4873473] [ !! ] Error 0xe8000022: The service is invalid. AMDeviceSecureStartService(device, serviceName, NULL, &dbgServiceConnection)
Could not run build/ios/iphoneos/Runner.app on aeb0dd50cf0b173f23b83dc325189a6bd1b9d2e6.
Try launching Xcode and selecting "Product > Run" to fix the problem:
  open iOS/Runner.xcworkspace

Running from Xcode gave the following error:

Failed to start remote service "com.apple.debugserver" on device.
Domain: com.apple.dtdevicekit
Code: 811
Recovery Suggestion: Please check your connection to your device.
User Info: {
    DVTRadarComponentKey = 261622;
}
--
The service is invalid.
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402653150
User Info: {
    DVTRadarComponentKey = 261622;
    MobileDeviceErrorCode = "(0xE8000022)";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x00000001223e699f DTDKCreateNSErrorFromAMDErrorCode + 220
	1   DTDeviceKitBase                     0x00000001223f6b94 __63-[DTDKRemoteDeviceConnection startFirstServiceOf:unlockKeybag:]_block_invoke + 613
	2   DTDeviceKitBase                     0x00000001223f6260 __48-[DTDKRemoteDeviceConnection futureWithSession:]_block_invoke_3 + 22
	3   DTDeviceKitBase                     0x00000001223e8aff __DTDKExecuteInSession_block_invoke_2 + 35
	4   DTDeviceKitBase                     0x00000001223e81d9 __DTDKExecuteWithConnection_block_invoke_2 + 218
	5   DTDeviceKitBase                     0x00000001223e80d6 __DTDKExecuteWithConnection_block_invoke + 106
	6   libdispatch.dylib                   0x00007fff6ed96658 _dispatch_client_callout + 8
	7   libdispatch.dylib                   0x00007fff6eda26ec _dispatch_lane_barrier_sync_invoke_and_complete + 60
	8   DVTFoundation                       0x000000010584cdf3 DVTDispatchBarrierSync + 208
	9   DVTFoundation                       0x0000000105823ea6 -[DVTDispatchLock performLockedBlock:] + 60
	10  DTDeviceKitBase                     0x00000001223e7fd7 DTDKExecuteWithConnection + 226
	11  DTDeviceKitBase                     0x00000001223e89a3 DTDKExecuteInSession + 239
	12  DTDeviceKitBase                     0x00000001223f60b2 __48-[DTDKRemoteDeviceConnection futureWithSession:]_block_invoke_2 + 131
	13  DVTFoundation                       0x000000010584a2ae __DVT_CALLING_CLIENT_BLOCK__ + 7
	14  DVTFoundation                       0x000000010584bed0 __DVTDispatchAsync_block_invoke + 1191
	15  libdispatch.dylib                   0x00007fff6ed956c4 _dispatch_call_block_and_release + 12
	16  libdispatch.dylib                   0x00007fff6ed96658 _dispatch_client_callout + 8
	17  libdispatch.dylib                   0x00007fff6ed9bc44 _dispatch_lane_serial_drain + 597
	18  libdispatch.dylib                   0x00007fff6ed9c609 _dispatch_lane_invoke + 414
	19  libdispatch.dylib                   0x00007fff6eda5c09 _dispatch_workloop_worker_thread + 596
	20  libsystem_pthread.dylib             0x00007fff6eff0a3d _pthread_wqthread + 290
	21  libsystem_pthread.dylib             0x00007fff6efefb77 start_wqthread + 15
);
}
--

System Information

macOS Version 10.15.7 (Build 19H15)
Xcode 12.2 (17535) (Build 12B45b)
Timestamp: 2020-11-22T12:10:35-05:00

```
</summary>

```
```

```
flutter analyze
Analyzing bug...                                               
No issues found! (ran in 1.8s)
```

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.4, on Mac OS X 10.15.7 19H15 darwin-x64, locale en-CA)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc4)
[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
[✓] Android Studio (version 3.6)
[!] IntelliJ IDEA Community Edition (version 2017.2.5)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.51.1)
[✓] Connected device (1 available)
```

</details>

**Comment by darshankawar** [comment]
@sowens-csd 
Can you provide complete `flutter doctor -v` ? Also with latest Xcode, the minimum ipad simulator version we get is 8th gen. If I am not wrong, you are trying to run on a 6th gen ipad. Can you try the same using an iPad simulator (8th gen) and check if you get similar error ?

**Comment by sowens-csd** [comment]
Unfortunately the simulators no longer offer 14.1 as a target since it has been updated, only 14.2. I can create a 6th gen iPad but not with that iOS version. 

Here's the -v output:

[✓] Flutter (Channel stable, 1.22.4, on Mac OS X 10.15.7 19H15 darwin-x64, locale en-CA)
    • Flutter version 1.22.4 at /Users/user.name/Documents/dev/flutter/sdk/flutter
    • Framework revision 1aafb3a8b9 (12 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc4)
    • Android SDK at /Users/user.name/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0-rc4
    • ANDROID_HOME = /Users/user.name/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.9.3

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[!] IntelliJ IDEA Community Edition (version 2017.2.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • iPad - App - 12.2 (mobile) • XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX • ios • com.apple.CoreSimulator.SimRuntime.iOS-12-2 (simulator)

! Doctor found issues in 1 category.

**Comment by darshankawar** [comment]
With the config I've, was able to run a sample app on iPad 8th gen (OS 14), since I don't have a physical or earlier version of iPad.

<img width="556" alt="Screenshot 2020-11-26 at 12 08 03 PM" src="https://user-images.githubusercontent.com/67046386/100316371-68a8ba00-2fe0-11eb-9477-d6c7074c3462.png">

Also checkout [this](https://github.com/ios-control/ios-deploy/issues/469#issuecomment-678552777) issue comment for reference. May be with the config you've for Xcode and iPad version.
Keeping this issue open for further analysis.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.4 19E2269 x86_64,
    locale en-IN)
    • Flutter version 1.23.0-18.1.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 198df796aa (6 weeks ago), 2020-10-15 12:04:33 -0700
    • Engine revision 1d12d82d9c
    • Dart version 2.11.0 (build 2.11.0-213.1.beta)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (5 available)
    • sdk gphone x86 arm (mobile)         • emulator-5554
      • android-x86    • Android 11 (API 30) (emulator)
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • iPad (8th generation) (mobile)      • E5E8E786-70A7-417B-A2DE-7074262D52E3
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • Web Server (web)                    • web-server
      • web-javascript • Flutter Tools
    • Chrome (web)                        • chrome
      • web-javascript • Google Chrome 87.0.4280.67

• No issues found!


```

</details>

**Comment by Hixie** [comment]
@xster Any chance you can try to repro this?

**Comment by xster** [comment]
I don't think you can reproduce this on simulators since we don't use ios-deploy.

@sowens-csd while you're here, can you run `ls /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/DeviceSupport`? 

That error generally means your Xcode doesn't have the iOS version's disk image support to be able to debug. Typically happens when the iOS version is higher than the Xcode version. I imagine the reverse is plausible. 

![Screen Shot 2020-12-01 at 9 06 29 PM](https://user-images.githubusercontent.com/156888/100830698-15d46580-3419-11eb-81ee-5f391cd1ac88.png)

Also try opening the Devices and Simulators window in Xcode (shift-cmd-2) and press the + button. See if starts downloading anything. Though likely nothing interesting will happen if you already upgraded your device to 14.2.

**Comment by sowens-csd** [comment]
```
ls /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/DeviceSupport
10.0 10.1 10.2 10.3 11.0 11.1 11.2 11.3 11.4 12.0 12.1 12.2 12.3 12.4 13.0 13.1 13.2 13.3 13.4 13.5 13.6 13.7 14.0 14.1 14.2 9.0  9.1  9.2  9.3
```

I tried pressing plus in the Devices and Simulators window and no, it did not start downloading anything. 14.2 was available on the drop down for OS. I like your theory about a mismatch between Xcode and iOS version. If I remember correctly my Xcode had already upgraded before the issue happened.

**Comment by xster** [comment]
Is the issue still reproducible or only before you updated the iPad from 14.1?

**Comment by jmagman** [comment]
@sowens-csd Did you do what the error message suggested?  Did it launch?
```
Try launching Xcode and selecting "Product > Run" to fix the problem:
  open ios/Runner.xcworkspace
```

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
